components/inputs/input-date.js

```diff
@@ -165,8 +176,14 @@ class InputDate extends SkeletonMixin(FormElementMixin(LocalizeCoreElement(LitEl
 		this._formattedValue = this.emptyText ? this.emptyText : '';
 
 		await (document.fonts ? document.fonts.ready : Promise.resolve());
-		const width = Math.ceil(parseFloat(getComputedStyle(this.shadowRoot.querySelector('.d2l-input-date-hidden-content')).getPropertyValue('width')));
-		this._hiddenContentWidth = `${width}px`;
+
+		const hiddenContent = this.shadowRoot.querySelector('.d2l-input-date-hidden-content');
+		this._hiddenContentResizeObserver = this._hiddenContentResizeObserver || new ResizeObserver(() => {
+			const width = Math.ceil(parseFloat(getComputedStyle(hiddenContent).getPropertyValue('width')));
+			this._hiddenContentWidth = `${width}px`;
+		});
+		this._hiddenContentResizeObserver.disconnect();
```

Comment:
Did you intend to disconnect right away?

Reply:
That's what I had seen done in other places but looking closer they were handling the case where the observer could exist already so it does seem unnecessary here.